### PR TITLE
Feature - Migrate more types from the app

### PR DESCRIPTION
**Pull request by e1himself**
Needed for https://github.com/prezly/prezly/pull/11668

### CODE REVIEW

File: src/endpoints/CampaignRecipients/scopes.ts [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
 
 export interface CampaignRecipientsScope {
-    type: 'scope:campaign_recipients';
+    type: ContactsScope.Type.CAMPAIGN_RECIPIENTS;
     campaign_id: Campaign['id'];
     selection?: SelectionValue<EmailRecipient['id']>;
 }
 
 export interface CampaignReportScope {
-    type: 'scope:campaign_recipients';
```
Comment (e1himself) [inline src/endpoints/CampaignRecipients/scopes.ts:27]: Fun fact: this was using a wrong `type`

File: src/endpoints/CampaignRecipients/scopes.ts [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
 
 export interface ContactOrganisationsScope {
-    type: 'scope:contact_organisations';
+    type: ContactsScope.Type.CONTACT_ORGANISATIONS;
     contact_id: Contact['id'];
     selection?: SelectionValue<Contact['id']>;
 }
 
 export interface ContactEmployeesScope {
-    type: 'scope:contact_organisations';
```
Comment (e1himself) [inline src/endpoints/CampaignRecipients/scopes.ts:15]: Fun fact: this was using a wrong `type`

File: src/types/Contact.ts [hunk 3]
```diff
@@ -3,8 +3,9 @@ import type { UploadedImage } from '@prezly/uploads';
 import type { ContactDuplicateSuggestion } from './ContactDuplicateSuggestion';
 import type { Entity } from './Entity';
 
-export interface ContactRef extends Entity {
-    contact_type: 'person' | 'organisation';
+export interface ContactRef extends Entity<number> {
+    id: number;
```
Comment (oleg-semyonov) [inline src/types/Contact.ts:7]: Is this needed, if `id` property is inherited from the `Entity` type? Same on line 20

**Review by OskarGasiorowski — APPROVED** [review]

File: src/types/License.ts [hunk 4]
```diff
@@ -9,16 +11,99 @@ export interface LicenseRef {
...
+    canceled_at: string | null;
+    company_address: string;
+    company_address_city: string;
+    company_address_country: string;
+    company_address_zip: string;
     created_at: string;
+    /**
+     * When there are no subscriptions yet, the currency is null (we do not set a default value).
+     */
+    currency: 'eur' | 'usd' | 'gbp' | null;
```
Comment (oleg-semyonov) [inline src/types/License.ts:39]: This looks like something that could be extracted into an enum :)

File: src/types/Entity.ts [hunk 5]
```diff
@@ -1,3 +1,3 @@
-export interface Entity<T extends number | string = number> {
+export interface Entity<T extends number | string = string | number> {
```
Comment (oleg-semyonov) [inline src/types/Entity.ts:1]: I don't see any `Entity` extends without specified type in this PR, which makes me a bit confused about this change.

Do we even have `string` IDs anywhere? 🤔 
I was under impression that all `id` fields are numbers, with strings left for `uuid`.

Comment (e1himself) [reply src/types/Entity.ts:1]: @oleg-semyonov Entity<string> is relied on in the main app. Also, `EmailRecipient` is de facto extending `Entity<string>`, though this is not reflected in the declaration (I'll add it, to be clear).

Thanks!

File: src/types/License.ts (hunk 4 above)
Comment (e1himself) [reply src/types/License.ts:39]: Will do. Thanks!

File: src/types/Contact.ts (hunk 3 above)
Comment (e1himself) [reply src/types/Contact.ts:7]: Well, it's not really necessary. But it also makes things more clear -- it's immediately obvious that there's `id` property. So I'm hesitant to remove it.

File: src/types/License.ts (hunk 4 above)
Comment (e1himself) [reply src/types/License.ts]: 434d8bd7faf939eb99a05999d972f4690a433cf9

File: src/types/Entity.ts (hunk 5 above)
Comment (e1himself) [reply src/types/Entity.ts:1]: 90212f8f53407bda0c92b022dff8994351e3d541

Comment (e1himself) [reply src/types/Entity.ts:1]: Hm... I'm now thinking maybe we don't even need `Entity` type in this repository. It's not adding anything to *this* codebase.

**Review by MohammadxAli — APPROVED** [review]

File: src/types/Entity.ts (hunk 5 above)
Comment (e1himself) [reply src/types/Entity.ts]: > I'm now thinking maybe we don't even need Entity type in this repository. It's not adding anything to this codebase.

Dropped with 85323871ac9888a597cc9a9e67fa41597ff8ff34

File: src/types/Contact.ts (hunk 3 above)
Comment (oleg-semyonov) [reply src/types/Contact.ts:6]: I was about to say that in that case you might as well ditch the `Entity` interface, but I see you already did :)

**Pull request merged**